### PR TITLE
XAUTOCLAIM Command

### PR DESCRIPTION
**Pull request by s-sathish**
XAUTOCLAIM Command impl in Jedis - https://github.com/redis/jedis/issues/2387

Notes -
1. New builder called 'STREAM_AUTO_CLAIM' is used to build the response from the xautoclaim command reply from redis. Reason is, it differs from the output of xclaim i.e the output of xautoclaim has two elements in the array -> 
streamId which can be used as the start for next call & list of successfully claimed entries.
2. A response object 'StreamAutoClaim' is used to store & access the response of xautoclaim command.

### CODE REVIEW

File: src/main/java/redis/clients/jedis/StreamAutoClaim.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package redis.clients.jedis;
+
+import java.io.Serializable;
+import java.util.List;
+
+public class StreamAutoClaim implements Serializable {
```
Comment (dengliming) [inline src/main/java/redis/clients/jedis/StreamAutoClaim.java]: What about naming `StreamClaimedMessages`?

File: src/main/java/redis/clients/jedis/commands/JedisClusterCommands.java [hunk 2]
```diff
@@ -634,5 +635,16 @@ default String setex(String key, int seconds, String value) {
   List<StreamEntryID> xclaimJustId(String key, String group, String consumername, long minIdleTime,
       XClaimParams params, StreamEntryID... ids);
 
+  /**
+   *  XAUTOCLAIM key group consumer min-idle-time start [COUNT count] [JUSTID]
+   */
+  StreamAutoClaim xautoclaim(String key, String group, String consumerName, long minIdleTime, StreamEntryID start);
```
Comment (dengliming) [inline src/main/java/redis/clients/jedis/commands/JedisClusterCommands.java]: Missing arg `JUSTID ` in `XAUTOCLAIM`. You can add a new method `xautoClaimJustId`

File: src/main/java/redis/clients/jedis/StreamAutoClaim.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+  private static final long serialVersionUID = 1L;
+
+  private StreamEntryID id;
+  private List<StreamEntry> streamEntries;
```
Comment (dengliming) [inline src/main/java/redis/clients/jedis/StreamAutoClaim.java]:
```suggestion
  private final StreamEntryID id;
  private final List<StreamEntry> streamEntries;
```

File: src/main/java/redis/clients/jedis/BuilderFactory.java [hunk 4]
```diff
@@ -802,6 +802,50 @@ public String toString() {
...
+        return null;
+      }
+
+      List<ArrayList<Object>> objectList = (List<ArrayList<Object>>) data;
+      if (objectList.isEmpty()) {
+        return null;
+      }
+
+      StreamEntryID nextStartEntryID = STREAM_ENTRY_ID.build(objectList.get(0));
+      objectList.remove(0);
```
Comment (dengliming) [inline src/main/java/redis/clients/jedis/BuilderFactory.java]: I think we use `index of list` from `1` directly would be better and no need to call `remove`.

**Review by dengliming — CHANGES_REQUESTED** [review]

File: src/main/java/redis/clients/jedis/StreamAutoClaim.java (hunk 1 above)
Comment (s-sathish) [reply src/main/java/redis/clients/jedis/StreamAutoClaim.java]: Renamed to StreamClaimedMessages

File: src/main/java/redis/clients/jedis/commands/JedisClusterCommands.java (hunk 2 above)
Comment (s-sathish) [reply src/main/java/redis/clients/jedis/commands/JedisClusterCommands.java]: Added support for JUSTID arg in XAUTOCLAIM command.

File: src/main/java/redis/clients/jedis/StreamAutoClaim.java (hunk 3 above)
Comment (s-sathish) [reply src/main/java/redis/clients/jedis/StreamAutoClaim.java]: Added final modifier to the fields

File: src/main/java/redis/clients/jedis/BuilderFactory.java (hunk 4 above)
Comment (s-sathish) [reply src/main/java/redis/clients/jedis/BuilderFactory.java]: Yes. Removed the remove(0) statement and used subList from index 1

File: src/main/java/redis/clients/jedis/BuilderFactory.java [hunk 5]
```diff
@@ -802,6 +802,82 @@ public String toString() {
...
+
+        Iterator<byte[]> hashIterator = hash.iterator();
+        Map<String, String> map = new HashMap<>(hash.size() / 2);
+        while (hashIterator.hasNext()) {
+          map.put(SafeEncoder.encode(hashIterator.next()), SafeEncoder.encode(hashIterator.next()));
+        }
+        streamEntries.add(new StreamEntry(entryID, map));
+      }
+
+      return new StreamClaimedMessages(nextStartEntryID, streamEntries);
```
Comment (dengliming) [inline src/main/java/redis/clients/jedis/BuilderFactory.java]:
```
List<Object> objectList = (List<Object>) data;
String minId = BuilderFactory.STRING.build(objectList.get(0));
return new StreamClaimedMessages(new StreamEntryID(minId), STREAM_ENTRY_LIST.build(objectList.get(1)));
```

File: src/main/java/redis/clients/jedis/BuilderFactory.java [hunk 6]
```diff
@@ -802,6 +802,82 @@ public String toString() {
...
+      List<StreamEntryID> streamEntryIDs = new ArrayList<>();
+
+      for (ArrayList<Object> res : objectList.subList(1, objectList.size())) {
+        if (res == null || res.isEmpty()) {
+          continue;
+        }
+        streamEntryIDs.add(STREAM_ENTRY_ID.build(res.get(0)));
+      }
+
+      return new StreamClaimedMessagesId(nextStartEntryID, streamEntryIDs);
```
Comment (dengliming) [inline src/main/java/redis/clients/jedis/BuilderFactory.java]:
```
List<Object> objectList = (List<Object>) data;
String minId = BuilderFactory.STRING.build(objectList.get(0));
return new StreamClaimedMessagesId(new StreamEntryID(minId), STREAM_ENTRY_ID_LIST.build(objectList.get(1)));
```

**Review by dengliming — CHANGES_REQUESTED** [review]

File: src/main/java/redis/clients/jedis/BuilderFactory.java (hunk 5 above)
Comment (s-sathish) [reply src/main/java/redis/clients/jedis/BuilderFactory.java]: Thanks for the review. Simplified the builder functions to make use of the existing functions

File: src/main/java/redis/clients/jedis/BuilderFactory.java (hunk 6 above)
Comment (s-sathish) [reply src/main/java/redis/clients/jedis/BuilderFactory.java]: Thanks for the review. Simplified the builder functions to make use of the existing functions

**Comment by s-sathish** [comment]
Hi @dengliming @sazzad16 I have resolved all the requested changes.
Kindly review once and let me know if there are any other changes to be made.

**Review by sazzad16 — CHANGES_REQUESTED** [review]
1. Four new methods seem too many to me. IMHO, we should use Params, which would reduce it to two methods.

    XAUTOCLAIM is provided as a convenience to the combination of XPENDING and XCLAIM. Any argument from those two commands may get into XAUTOCLAIM. Params would be helpful in such case.

2. We may skip `StreamClaimedMessages` and `StreamClaimedMessagesId`.
    `Map.Entry<StreamEntryID, List<StreamEntryID/StreamEntry>>` should be enough.

File: src/main/java/redis/clients/jedis/BinaryClient.java [hunk 7]
```diff
@@ -1977,6 +1977,34 @@ public void xclaimJustId(byte[] key, byte[] groupname, byte[] consumername, long
     xclaim(key, groupname, consumername, minIdleTime, params, ids, true);
   }
 
+  public void xautoclaim(byte[] key, byte[] groupName, byte[] consumerName,
+      long minIdleTime, byte[] start, XAutoClaimParams params) {
+    xautoclaim(key, groupName, consumerName, minIdleTime, start, params, false);
+  }
+
+  public void xautoclaim(byte[] key, byte[] groupName, byte[] consumerName,
+      long minIdleTime, byte[] start, XAutoClaimParams params, boolean justId) {
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/BinaryClient.java:1986]:
```suggestion
  private void xautoclaim(byte[] key, byte[] groupName, byte[] consumerName,
      long minIdleTime, byte[] start, XAutoClaimParams params, boolean justId) {
```

File: src/main/java/redis/clients/jedis/commands/JedisCommands.java [hunk 8]
```diff
@@ -650,6 +651,18 @@ default String setex(String key, int seconds, String value) {
   List<StreamEntryID> xclaimJustId(String key, String group, String consumername, long minIdleTime,
       XClaimParams params, StreamEntryID... ids);
 
+  /**
+   * XAUTOCLAIM key group consumer min-idle-time start [COUNT count] [JUSTID]
+   */
+  Map.Entry<StreamEntryID, List<StreamEntry>> xautoclaim(String key, String group, String consumerName,
+      long minIdleTime, StreamEntryID start, XAutoClaimParams params);
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/commands/JedisCommands.java:658]:
```suggestion
  /**
   * XAUTOCLAIM key group consumer min-idle-time start [COUNT count]
   */
  Map.Entry<StreamEntryID, List<StreamEntry>> xautoclaim(String key, String group, String consumerName,
      long minIdleTime, StreamEntryID start, XAutoClaimParams params);
```

File: src/main/java/redis/clients/jedis/commands/JedisCommands.java [hunk 9]
```diff
@@ -650,6 +651,18 @@ default String setex(String key, int seconds, String value) {
...
+
+  /**
+   * XAUTOCLAIM key group consumer min-idle-time start [COUNT count] [JUSTID]
+   */
+  Map.Entry<StreamEntryID, List<StreamEntryID>> xautoclaimJustId(String key, String group, String consumerName,
+      long minIdleTime, StreamEntryID start, XAutoClaimParams params);
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/commands/JedisCommands.java:664]:
```suggestion
  /**
   * XAUTOCLAIM key group consumer min-idle-time start [COUNT count] JUSTID
   */
  Map.Entry<StreamEntryID, List<StreamEntryID>> xautoclaimJustId(String key, String group, String consumerName,
      long minIdleTime, StreamEntryID start, XAutoClaimParams params);
```

File: src/main/java/redis/clients/jedis/commands/JedisClusterCommands.java [hunk 10]
```diff
@@ -634,5 +635,17 @@ default String setex(String key, int seconds, String value) {
...
+   *  XAUTOCLAIM key group consumer min-idle-time start [COUNT count] [JUSTID]
+   */
+  Map.Entry<StreamEntryID, List<StreamEntry>> xautoclaim(String key, String group, String consumerName,
+      long minIdleTime, StreamEntryID start, XAutoClaimParams params);
+
+  /**
+   *  XAUTOCLAIM key group consumer min-idle-time start [COUNT count] [JUSTID]
+   */
+  Map.Entry<StreamEntryID, List<StreamEntryID>> xautoclaimJustId(String key, String group, String consumerName,
+      long minIdleTime, StreamEntryID start, XAutoClaimParams params);
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/commands/JedisClusterCommands.java:648]: Check JedisCommands.

**Comment by s-sathish** [comment]
Hi @sazzad16  I have resolved all the requested changes.
1. Introduce `XAutoClaimParams` so that in the future if new options are added to the command, it will be easy to add it here as well.
2. Skipped `StreamClaimedMessages, StreamClaimedMessagesId` in favour of raw `Map.Entry<StreamEntryID, List<StreamEntry>>, Map.Entry<StreamEntryID, List<StreamEntryID>>` respectively

Kindly review once and let me know if there are any other changes to be made.

File: src/main/java/redis/clients/jedis/commands/BinaryJedisCommands.java [hunk 11]
```diff
@@ -450,6 +450,12 @@ default String setex(byte[] key, int seconds, byte[] value) {
 
   List<byte[]> xclaimJustId(byte[] key, byte[] group, byte[] consumername, long minIdleTime, XClaimParams params, byte[]... ids);
 
+  Object xautoclaim(byte[] key, byte[] groupName, byte[] consumerName,
+      long minIdleTime, byte[] start, XAutoClaimParams params);
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/commands/BinaryJedisCommands.java:454]:
```suggestion
  List<Object> xautoclaim(byte[] key, byte[] groupName, byte[] consumerName,
      long minIdleTime, byte[] start, XAutoClaimParams params);
```

File: src/main/java/redis/clients/jedis/commands/BinaryJedisCommands.java [hunk 12]
```diff
@@ -450,6 +450,12 @@ default String setex(byte[] key, int seconds, byte[] value) {
...
+
+  Object xautoclaimJustId(byte[] key, byte[] groupName, byte[] consumerName,
+      long minIdleTime, byte[] start, XAutoClaimParams params);
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/commands/BinaryJedisCommands.java:457]:
```suggestion
  List<Object> xautoclaimJustId(byte[] key, byte[] groupName, byte[] consumerName,
      long minIdleTime, byte[] start, XAutoClaimParams params);
```

File: src/main/java/redis/clients/jedis/commands/JedisCommands.java (hunk 9 above)
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/commands/JedisCommands.java:664]: In JavaDoc, please specify that `start` param can be `null` without any issue. (This is handled in Client class.)

PS: This is not true for binary class.

File: src/main/java/redis/clients/jedis/BuilderFactory.java [hunk 13]
```diff
@@ -802,6 +802,46 @@ public String toString() {
     }
   };
 
+  public static final Builder<Map.Entry<StreamEntryID, List<StreamEntry>>> STREAM_AUTO_CLAIM
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/BuilderFactory.java:805]: Add a recognizable suffix with the variable name.

File: src/main/java/redis/clients/jedis/BuilderFactory.java [hunk 14]
```diff
@@ -802,6 +802,46 @@ public String toString() {
...
+          STREAM_ENTRY_LIST.build(objectList.get(1)));
+    }
+
+    @Override
+    public String toString() {
+      return "Map.Entry<StreamEntryID, List<StreamEntry>>";
+    }
+  };
+
+  public static final Builder<Map.Entry<StreamEntryID, List<StreamEntryID>>> STREAM_AUTO_CLAIM_ID
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/BuilderFactory.java:825]: Add a recognizable suffix with the variable name.

File: src/main/java/redis/clients/jedis/BinaryJedis.java [hunk 15]
```diff
@@ -4939,6 +4939,22 @@ public Object xpending(final byte[] key, final byte[] groupname) {
     return client.getBinaryMultiBulkReply();
   }
 
+  @Override
+  public Object xautoclaim(byte[] key, byte[] groupName, byte[] consumerName,
+      long minIdleTime, byte[] start, XAutoClaimParams params) {
+    checkIsInMultiOrPipeline();
+    client.xautoclaim(key, groupName, consumerName, minIdleTime, start, params);
+    return client.getBinaryMultiBulkReply();
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/BinaryJedis.java:4947]: Not sure, but may be?
```suggestion
    return client.getObjectMultiBulkReply();
```

File: src/main/java/redis/clients/jedis/BinaryJedis.java [hunk 16]
```diff
@@ -4939,6 +4939,22 @@ public Object xpending(final byte[] key, final byte[] groupname) {
...
+  }
+
+  @Override
+  public Object xautoclaimJustId(byte[] key, byte[] groupName, byte[] consumerName,
+      long minIdleTime, byte[] start, XAutoClaimParams params) {
+    checkIsInMultiOrPipeline();
+    client.xautoclaimJustId(key, groupName, consumerName, minIdleTime, start, params);
+    return client.getBinaryMultiBulkReply();
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/BinaryJedis.java:4955]: Not sure, but may be?
```suggestion
    return client.getObjectMultiBulkReply();
```

**Review by sazzad16 — CHANGES_REQUESTED** [review]

File: src/main/java/redis/clients/jedis/BuilderFactory.java (hunk 13 above)
Comment (s-sathish) [reply src/main/java/redis/clients/jedis/BuilderFactory.java:805]: Hi @sazzad16 Can you please clarify here ? I didn't get what you suggested here.
`STREAM_AUTO_CLAIM, STREAM_AUTO_CLAIM_ID` are the builders for xautoclaim & xautoclaimJustId commands respectively.

Comment (sazzad16) [reply src/main/java/redis/clients/jedis/BuilderFactory.java:805]: @s-sathish Almost all the builders here in BuilderFactory represent an (dedicated) object/collection of (dedicated) object. In other cases, we should make a separation. Only thing here we can separate is the variable name. This what I'm asking here.

I.e. you can use `STREAM_AUTO_CLAIM_RESPONSE` and `STREAM_AUTO_CLAIM_ID_RESPONSE` as names.

Comment (sazzad16) [reply src/main/java/redis/clients/jedis/BuilderFactory.java:805]: @s-sathish Almost all the builders here in BuilderFactory represent an (dedicated) object/collection of (dedicated) object. In other cases, we should make a separation. Only thing here we can separate is the variable name. This is what I'm asking here.

I.e. you can use `STREAM_AUTO_CLAIM_RESPONSE` and `STREAM_AUTO_CLAIM_ID_RESPONSE` as names.

File: src/main/java/redis/clients/jedis/BinaryClient.java (hunk 7 above)
Comment (s-sathish) [reply src/main/java/redis/clients/jedis/BinaryClient.java]: Changed the access specifier to private for this internal method

File: src/main/java/redis/clients/jedis/commands/JedisCommands.java (hunk 8 above)
Comment (s-sathish) [reply src/main/java/redis/clients/jedis/commands/JedisCommands.java]: JavaDoc changes done

File: src/main/java/redis/clients/jedis/commands/JedisCommands.java (hunk 9 above)
Comment (s-sathish) [reply src/main/java/redis/clients/jedis/commands/JedisCommands.java]: JavaDoc changes done

File: src/main/java/redis/clients/jedis/commands/JedisClusterCommands.java (hunk 10 above)
Comment (s-sathish) [reply src/main/java/redis/clients/jedis/commands/JedisClusterCommands.java]: JavaDoc changes done

File: src/main/java/redis/clients/jedis/BuilderFactory.java (hunk 13 above)
Comment (s-sathish) [reply src/main/java/redis/clients/jedis/BuilderFactory.java]: Got it. Changes done.

File: src/main/java/redis/clients/jedis/BuilderFactory.java (hunk 14 above)
Comment (s-sathish) [reply src/main/java/redis/clients/jedis/BuilderFactory.java]: Changed the variable name.

**Comment by s-sathish** [comment]
Hi @sazzad16 The requested changes are done. Kindly check once.

One test case failed in the latest merge commit i.e I merged the latest master to my branch. 
I believe that test case (JedisTest - timeoutConnection) failure was intermittent ?
How can we re-run the build ?

File: src/main/java/redis/clients/jedis/BuilderFactory.java [hunk 17]
```diff
@@ -802,6 +802,46 @@ public String toString() {
...
+    @Override
+    @SuppressWarnings("unchecked")
+    public Map.Entry<StreamEntryID, List<StreamEntry>> build(Object data) {
+      if (null == data) {
+        return null;
+      }
+
+      List<ArrayList<Object>> objectList = (List<ArrayList<Object>>) data;
+      return new AbstractMap.SimpleEntry<>(new StreamEntryID(BuilderFactory.STRING.build(objectList.get(0))),
+          STREAM_ENTRY_LIST.build(objectList.get(1)));
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/BuilderFactory.java:816]:
```suggestion
      return new AbstractMap.SimpleEntry<>(STREAM_ENTRY_ID.build(objectList.get(0)),
          STREAM_ENTRY_LIST.build(objectList.get(1)));
```

File: src/main/java/redis/clients/jedis/BuilderFactory.java [hunk 18]
```diff
@@ -802,6 +802,46 @@ public String toString() {
...
+    @Override
+    @SuppressWarnings("unchecked")
+    public Map.Entry<StreamEntryID, List<StreamEntryID>> build(Object data) {
+      if (null == data) {
+        return null;
+      }
+
+      List<ArrayList<Object>> objectList = (List<ArrayList<Object>>) data;
+      return new AbstractMap.SimpleEntry<>(new StreamEntryID(BuilderFactory.STRING.build(objectList.get(0))),
+          STREAM_ENTRY_ID_LIST.build(objectList.get(1)));
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/BuilderFactory.java:836]:
```suggestion
      return new AbstractMap.SimpleEntry<>(STREAM_ENTRY_ID.build(objectList.get(0)),
          STREAM_ENTRY_ID_LIST.build(objectList.get(1)));
```

File: src/main/java/redis/clients/jedis/commands/BinaryRedisPipeline.java [hunk 19]
```diff
@@ -423,6 +423,12 @@
   Response<List<byte[]>> xclaimJustId(byte[] key, byte[] group, byte[] consumername,
       long minIdleTime, XClaimParams params, byte[]... ids);
 
+  Response<Object> xautoclaim(byte[] key, byte[] group, byte[] consumerName,
+      long minIdleTime, byte[] start, XAutoClaimParams params);
+
+  Response<Object> xautoclaimJustId(byte[] key, byte[] group, byte[] consumerName,
+      long minIdleTime, byte[] start, XAutoClaimParams params);
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/commands/BinaryRedisPipeline.java:430]:
```suggestion
  Response<List<Object>> xautoclaim(byte[] key, byte[] group, byte[] consumerName,
      long minIdleTime, byte[] start, XAutoClaimParams params);

  Response<List<Object>> xautoclaimJustId(byte[] key, byte[] group, byte[] consumerName,
      long minIdleTime, byte[] start, XAutoClaimParams params);
```

File: src/main/java/redis/clients/jedis/commands/BinaryJedisClusterCommands.java [hunk 20]
```diff
@@ -421,6 +421,12 @@ default String setex(byte[] key, int seconds, byte[] value) {
   List<byte[]> xclaimJustId(byte[] key, byte[] group, byte[] consumername, long minIdleTime,
       XClaimParams params, byte[]... ids);
 
+  Object xautoclaim(byte[] key, byte[] groupName, byte[] consumerName,
+      long minIdleTime, byte[] start, XAutoClaimParams params);
+
+  Object xautoclaimJustId(byte[] key, byte[] groupName, byte[] consumerName,
+      long minIdleTime, byte[] start, XAutoClaimParams params);
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/commands/BinaryJedisClusterCommands.java:428]:
```suggestion
  List<Object> xautoclaim(byte[] key, byte[] groupName, byte[] consumerName,
      long minIdleTime, byte[] start, XAutoClaimParams params);

  List<Object> xautoclaimJustId(byte[] key, byte[] groupName, byte[] consumerName,
      long minIdleTime, byte[] start, XAutoClaimParams params);
```

**Review by sazzad16 — CHANGES_REQUESTED** [review]

File: src/main/java/redis/clients/jedis/BuilderFactory.java [hunk 21]
```diff
@@ -802,6 +802,46 @@ public String toString() {
...
+  public static final Builder<Map.Entry<StreamEntryID, List<StreamEntry>>> STREAM_AUTO_CLAIM_RESPONSE
+      = new Builder<Map.Entry<StreamEntryID, List<StreamEntry>>>() {
+    @Override
+    @SuppressWarnings("unchecked")
+    public Map.Entry<StreamEntryID, List<StreamEntry>> build(Object data) {
+      if (null == data) {
+        return null;
+      }
+
+      List<ArrayList<Object>> objectList = (List<ArrayList<Object>>) data;
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/BuilderFactory.java:814]:
```suggestion
      List<Object> objectList = (List<Object>) data;
```

File: src/main/java/redis/clients/jedis/BuilderFactory.java [hunk 22]
```diff
@@ -802,6 +802,46 @@ public String toString() {
...
+  public static final Builder<Map.Entry<StreamEntryID, List<StreamEntryID>>> STREAM_AUTO_CLAIM_ID_RESPONSE
+      = new Builder<Map.Entry<StreamEntryID, List<StreamEntryID>>>() {
+    @Override
+    @SuppressWarnings("unchecked")
+    public Map.Entry<StreamEntryID, List<StreamEntryID>> build(Object data) {
+      if (null == data) {
+        return null;
+      }
+
+      List<ArrayList<Object>> objectList = (List<ArrayList<Object>>) data;
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/BuilderFactory.java:834]:
```suggestion
      List<Object> objectList = (List<Object>) data;
```

**Review by sazzad16 — CHANGES_REQUESTED** [review]

**Comment by s-sathish** [comment]
@sazzad16 Need your help here in re-running the last commit test build.
Looks like it was an intermittent issue.
From the test build logs -
`JedisClusterTest.testMigrate:324 » JedisCluster CLUSTERDOWN The cluster is dow...`

**Review by sazzad16 — APPROVED** [review]

**Review by dengliming — APPROVED** [review]

**Comment by sazzad16** [comment]
Merged in master. Pending backport to release Jedis 3.6.0.

Thanks @s-sathish for contribution!

**Pull request merged**